Add cortex_frontend_query_result_cache_skipped_total  metric (#2855)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Add sanity check at startup to ensure the configured filesystem directories don't overlap for different components. #2828
* [ENHANCEMENT] Go: updated to go 1.19.1. #2637
* [ENHANCEMENT] Runtime config: don't unmarshal runtime configuration files if they haven't changed. This can save a bit of CPU and memory on every component using runtime config. #2954

> * [ENHANCEMENT] Query-frontend: Add `cortex_frontend_query_result_cache_skipped_total` metric to track the reason why a query results is not cached. #2855

Reviewer: ```suggestion
* [ENHANCEMENT] Query-frontend: Add `cortex_frontend_query_result_cache_skipped_total` metric to track the reason why query results are not cached. #2855
```
